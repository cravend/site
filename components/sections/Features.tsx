import { useTranslations } from "next-intl";

type FeatureType = "education";

function Feature({ type }: { type: FeatureType }) {
  const t = useTranslations(`Features.${type}`);

  return (
    // TODO: figure out semantic html
    <div>
      <h2>{t("header")}</h2>
      <p>{t.rich("content")}</p>
    </div>
  );
}

function Features() {}

export default Features;
