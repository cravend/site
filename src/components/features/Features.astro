---
import { t } from "astro-i18n";
import Feature from "./Feature.astro";
---

<!-- TODO: figure out grid -->
<section>
  <h2 class="sr-only">{t("Features.header")}</h2>
  <Feature header={t("Features.education.header")}>
    <p set:html={t("Features.education.firstParagraph")} />
  </Feature>
  <Feature header={t("Features.study_abroad.header")}>
    <p set:html={t("Features.study_abroad.firstParagraph")} />
  </Feature>
  <Feature header={t("Features.work.header")}>
    <p set:html={t("Features.work.firstParagraph")} />
    <p set:html={t("Features.work.secondParagraph")} />
  </Feature>
  <Feature header={t("Features.freelance.header")}>
    <p set:html={t("Features.freelance.firstParagraph")} />
    <p set:html={t("Features.freelance.secondParagraph")} />
  </Feature>
</section>
