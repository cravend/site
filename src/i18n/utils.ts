import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
