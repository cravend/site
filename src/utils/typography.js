import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
