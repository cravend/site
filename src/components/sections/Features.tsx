import { useTranslations } from "next-intl";

import { useLocale } from "../../i18n/hooks";
import styles from "../../styles/modules/sections/Features.module.scss";
import Feature from "../Feature";
import Link from "../Link";

const RESUME_MAPPING = {
  en: "/resumes/Dalton Craven Résumé (en).pdf",
  fr: "/resumes/Dalton Craven Résumé (en).pdf",
  ar: "/resumes/Dalton Craven Résumé (en).pdf",
} as const;

const Features = () => {
  const t = useTranslations("sections.Features");
  const [locale] = useLocale();

  return (
    <div className={styles.grid}>
      <Feature heading={t.rich("education.header")}>
        <p>
          {t.rich("education.content", {
            Link: (children) => (
              <Link to="https://sc.edu/about/offices_and_divisions/undergraduate_admissions/honors_and_scholars_programs/top_scholars/index.php">
                {children}
              </Link>
            ),
          })}
        </p>
      </Feature>
      <Feature heading={t.rich("study_abroad.header")}>
        <p>{t.rich("study_abroad.content")}</p>
      </Feature>
      <Feature heading={t.rich("work.header")}>
        <p>
          {t.rich("work.content.intro", {
            Link: (text) => <Link to="https://qantev.com">{text}</Link>,
          })}
        </p>
        <p>{t.rich("work.content.update")}</p>
      </Feature>
      <Feature heading={t.rich("freelance.header")}>
        <p>{t.rich("freelance.content.intro")}</p>
        <ul>
          <li>
            <Link to="https://crystalclearcontractors.com">
              Crystal Clear Contracting
            </Link>
          </li>
          <li>
            <Link to="https://crystalclearconcrete.com">
              Crystal Clear Concrete
            </Link>
          </li>
          <li>
            <Link to="https://cleancrystalclear.com">Clean Crystal Clear</Link>
          </li>
          <li>
            <Link to="https://elitepaintcincy.com">Elite Painting</Link>
          </li>
          <li>
            <Link to="https://elitecontractingcincy.com">
              Elite Contracting
            </Link>
          </li>
        </ul>
        <p>
          {t.rich("freelance.content.conclusion", {
            Resume: (text) => (
              <Link locale={false} to={RESUME_MAPPING[locale]} openInNewTab>
                {text}
              </Link>
            ),
            Contact: (text) => <Link to="#contact">{text}</Link>,
          })}
        </p>
      </Feature>
    </div>
  );
};

export default Features;
