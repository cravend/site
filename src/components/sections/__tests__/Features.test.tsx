import { render } from "../../../tests/utils";
import Features from "../Features";

describe("<Features />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Features />);
    expect(container).toMatchSnapshot();
  });
});
