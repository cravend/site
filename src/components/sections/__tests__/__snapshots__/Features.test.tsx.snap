// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> renders 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="card"
    >
      <h2
        class="heading"
      >
        Education
      </h2>
      <p>
        I’m a senior at the University of South Carolina. I’m pursuing a Bachelor of Science in Computer Science with a major in computer science and a minor in psychology (weird, right?). I’m also a McNair Scholar as part of UofSC’s most prestigious scholarship program, the 
        <a
          class="primary"
          href="https://sc.edu/about/offices_and_divisions/undergraduate_admissions/honors_and_scholars_programs/top_scholars/index.php"
        >
          University Top Scholars
        </a>
        .
      </p>
    </div>
    <div
      class="card"
    >
      <h2
        class="heading"
      >
        Study Abroad
      </h2>
      <p>
        I have had multiple study- and work-abroad experiences while at university. I studied for both semesters of 2021 at the American University of Paris, weathering lockdowns & strikes while gaining valuable life experience. While at AUP, I completed coursework in both computer science & psychology, as well as in French language and culture.
      </p>
    </div>
    <div
      class="card"
    >
      <h2
        class="heading"
      >
        Work
      </h2>
      <p>
        In the summer of 2021, in between my two semesters of study abroad, I completed my fifth internship at 
        <a
          class="primary"
          href="https://qantev.com"
        >
          Qantev
        </a>
        , working primarily as a front-end developer on a React-based application. I also learned Python and Pandas to single-handedly work on the "full stack" of a new feature.
      </p>
      <p>
        I then returned to Qantev during summer 2022, focusing on migrating our codebase from Javascript to Typescript. Over my 3–month period, I coordinated the refactoring of 55% of our files and oversaw the implementation of new TypeScript-specific best practices.
      </p>
    </div>
    <div
      class="card"
    >
      <h2
        class="heading"
      >
        Freelance
      </h2>
      <p>
        I also do freelance web development! I work primarily with small businesses (although I’m open to other opportunities), with an emphasis on providing resources and training to ensure long-term functionality.
      </p>
      <ul>
        <li>
          <a
            class="primary"
            href="https://crystalclearcontractors.com"
          >
            Crystal Clear Contracting
          </a>
        </li>
        <li>
          <a
            class="primary"
            href="https://crystalclearconcrete.com"
          >
            Crystal Clear Concrete
          </a>
        </li>
        <li>
          <a
            class="primary"
            href="https://cleancrystalclear.com"
          >
            Clean Crystal Clear
          </a>
        </li>
        <li>
          <a
            class="primary"
            href="https://elitepaintcincy.com"
          >
            Elite Painting
          </a>
        </li>
        <li>
          <a
            class="primary"
            href="https://elitecontractingcincy.com"
          >
            Elite Contracting
          </a>
        </li>
      </ul>
      <p>
        To learn more about my work, feel free to visit my 
        <a
          class="primary"
          href="/resumes/Dalton%20Craven%20R%C3%A9sum%C3%A9%20(en).pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          résumé
        </a>
         or contact me at the 
        <a
          class="primary"
          href="/#contact"
        >
          links below
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
